spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -665,7 +758,80 @@ protected HeaderNames getHeaderNames() {
 	 */
 	public static class HeaderNames {
 
+		/**
+		 * Bits representing which headers to add.
+		 * @since 2.8.4
+		 */
+		public enum HeadersToAdd {
+
+			/**
+			 * The offset of the failed record.
+			 */
+			OFFSET(0),
```

Comment:
Looks like just an `ordinal()` is enough.
I mean all those explicit values for this `enum` is fully aligned with an `ordinal()` of the enum:
```

    /**
     * Returns the ordinal of this enumeration constant (its position
     * in its enum declaration, where the initial constant is assigned
     * an ordinal of zero).
     *
     * Most programmers will have no use for this method.  It is
     * designed for use by sophisticated enum-based data structures, such
     * as {@link java.util.EnumSet} and {@link java.util.EnumMap}.
     *
     * @return the ordinal of this enumeration constant
     */
    public final int ordinal() {
```